torch/csrc/distributed/c10d/TraceUtils.h

```diff
@@ -528,12 +528,13 @@ struct NCCLTraceBuffer {
   This is called by the watchdog thread, and is asynchronous from the
   perspective of the main thread.
 
-  compute_duration defaults to false since it requires calling the cuda APIs in
-  getDurationFromEvent which might increase the peak memory usage. Also
-  enableTiming_ automatically enabled if compute_duration is true for
-  computation to work correctly - see TORCH_NCCL_ENABLE_TIMING.
+  compute_duration defaults to true since retire_id is only called in the
+  watchdog thread, which is currently a place we call cuda APIs which may hang,
+  but care should be taken to avoid computing duration in any function that must
+  never hang. (timing must also be enabled for compute_duration - see
```

Comment:
this is no longer correct as of this PR landing, but if you're fixing it in a later PR I'm happy to land this as a pure revert

Reply:
ok looks like you aren't updating this later.  better just update it here probably.

Reply:
or maybe it is still correct, given the CI results of PRs later in the stack. I am going to land this as is and sort out what is happening in the CI for the later PRs.